action/tests/test_main.py

```diff
@@ -2,18 +2,21 @@
 
 # pylint: disable=use-dict-literal
 
-import re
+from __future__ import annotations
+
 import sys
 from contextlib import contextmanager
-from pathlib import Path
 from runpy import run_module
 from subprocess import PIPE, STDOUT, CompletedProcess  # nosec
 from types import SimpleNamespace
-from typing import Dict, Generator
+from typing import TYPE_CHECKING, Generator
 from unittest.mock import ANY, Mock, call, patch
 
 import pytest
 
+if TYPE_CHECKING:
+    from pathlib import Path
+
```

Comment:
Question: I keep seeing this pattern in some Python projects (not just in `darker`). Is there some clear benefit on conditional importing type hints? I've tried to avoid it since I ran into some problems with it in the past but maybe the situation has changes since then.

Reply:
Linters (e.g. `ruff`) insist on this. On one occasion putting the import outside the if clause would have caused an import loop in my code. And I guess in some cases unneeded imports might be avoided (in case the user never hits some particular functionality).